src/lib/operators/import.cpp

```diff
@@ -66,15 +70,25 @@ std::shared_ptr<const Table> Import::_on_execute() {
 
   Hyrise::get().storage_manager.add_table(_tablename, table);
 
-  // must match ImportNode::output_expressions
+  // If a table encoding is specified, encode the table accordingly.
+  // The default encoding is `EncodingType::Dictionary` except for binary files.
+  // For binary files the default is the encoding of the file.
+  auto encoding_to_apply =
+      (_file_type != FileType::Binary && !_table_encoding) ? EncodingType::Dictionary : _table_encoding;
+
+  if (encoding_to_apply) {
+    const auto chunk_encoding_spec = ChunkEncodingSpec(table->column_count(), SegmentEncodingSpec(*encoding_to_apply));
+    ChunkEncoder::encode_all_chunks(table, chunk_encoding_spec);
+  }
```

Comment:
Can't we just do the following (`_target_encoding` can even remain `const`)?
```suggestion
  if (_target_encoding || _file_type != FileType::Binary) {
    const auto encoding_spec = _target_encoding ? SegmentEncodingSpec{*_target_encoding} : {};
    ChunkEncoder::encode_all_chunks(table, encoding_spec);
  }
```